### ISSUE TITLE
Trouble visualizing the outputs

### ISSUE THREAD

**Issue by kristupas-g** [EricBrock/RSI-Net#2]
I have trained the model NWPU45 dataset just like you. But when I try to do inference and visualize the results, the outputs are very blue and it barely looks like the original image. I've clipped the outputs to be between 0 and 1 and also converted them to RGB format, but that didn't help.

**Comment by EricBrock** [comment]
Hello, may I ask what type of tasks you were trained on? How many samples were in your training dataset, and are you sure the parameters you were trained with are correct? What was your batch size set to? Also, could you provide information about your operating environment?

> I have trained the model NWPU45 dataset just like you. But when I try to do inference and visualize the results, the outputs are very blue and it barely looks like the original image. I've clipped the outputs to be between 0 and 1 and also converted them to RGB format, but that didn't help.

**Comment by kristupas-g** [comment]
I trained for the colorx2 task which If I understood correctly should produce colored images at twice the original scale and I have used the provided `train.py` script for training without changing anything. I'm also adding a screenshot of the output of the model to my reply. As for my environment I'm working in Google Colab
![image](https://github.com/EricBrock/RSI-Net/assets/78504062/77d70603-5867-4c6e-a956-5db3d7a3b44b)

**Comment by EricBrock** [comment]
How about your train datasets? I randomly selected 9920 images in 16 classes of remote sensing scenes as training samples, and 800 as test samples. Did you use all NWPU-45 datasets? As far as I am concerned, the datasets have a great influence on colorization tasks.

**Comment by kristupas-g** [comment]
For training I've used 9920 images and 800 for testing. I got these images by using `prepare.py` utility that is provided. I'm also adding the a gist to the source code that I've used for training, maybe that will give you some insight. This code was ran in a `Google Colab` environment. [Link to source code](https://gist.github.com/kristupas-g/700da31f9a26f358c9d85e1d682f8d89)

Edit: I've also done the exact same thing for the X2 task but got the same blueish result. I also tried to visualize the output with matplotlib, but that produced an even poorer image. Things I did before visualizing with matplotlib:
1. Convert to RGB format since model outputs BGR
2. Normalize the values to [0, 1] by `(model_output + 1) / 2`

**Comment by EricBrock** [comment]
I have noticed that your torch version is 2.0 which is not the same as I used. 
And I don't know why you use (model_output+1)/2 to normalize the value of output.
I also noticed that your model was not trained because the result of eval does not change at all.  I guess your environment is not suitable. Because I have used the same code on RTX 2080ti and got a similar result as trained on RTX 3090, they can both work.

**Comment by kristupas-g** [comment]
Thank you I will try training the model by copying the code to a code cell and running it instead maybe that will help. As to why I normalized the output like I did is because the model outputed values between `[-1, 1]`

**Comment by EricBrock** [comment]
I just normalize the values to [0, 1.] and you can ignore the part of [-1, 0].
In all codes, I only used OpenCV to read and save images, other packages are allowed if you use them correctly.
